hfc/fabric/client.py

```diff
@@ -127,6 +127,7 @@ def init_with_net_profile(self, profile_path='network.json'):
             peer.init_with_bundle(peers[name])
             self._peers[name] = peer
 
+    # TODO: make this function optional when starting the client.
     async def init_with_discovery(self, requestor, peer_target,
```

Comment:
I think we need to move this function back to `channel.py`. @Kelvin-M Sorry but I forgot why was it outside the channel? I checked the nodejs sdk but they put it under channel. Well, I am raising the issue because I received some feedbacks saying that if setting `channel_name` to None, there will be a lot of errors.

Reply:
@dexhunter I think I put it here because `init_with_net_profile` is there :)
Yes, a channel name should be mandatory, maybe we can remove the `channel_name=None` to fix partially those errors.
I think we will need to refacto the `client.py` and the `channel.py` along nodejs code !